R/combinedscatterplot.R

```diff
@@ -0,0 +1,827 @@
+                            y.tick.font.family = global.font.family,
+                            y.tick.font.size = 10,
+                            x.title = "",
+                            x.title.font.color = global.font.color,
+                            x.title.font.family = global.font.family,
+                            x.title.font.size = 12,
+                            x.line.width = 0,
+                            x.line.color = rgb(0, 0, 0, maxColorValue = 255),
+                            x.tick.mark.length = 3,
+                            x.tick.mark.color = "transparent",
+                            x.bounds.minimum = NULL,
+                            x.bounds.maximum = NULL,
+                            x.tick.distance = NULL,
+                            x.tick.maxnum = NULL,
+                            x.zero.line.width = 0,
+                            x.zero.line.color = rgb(225, 225, 225, maxColorValue = 255),
+                            x.grid.width = 1 * grid.show,
+                            x.grid.color = rgb(225, 225, 225, maxColorValue = 255),
+                            x.tick.show = TRUE,
+                            x.tick.suffix = "",
+                            x.tick.prefix = "",
+                            x.tick.format = "",
+                            x.hovertext.format = "",
+                            x.tick.angle = NULL,
+                            x.tick.font.color = global.font.color,
+                            x.tick.font.family = global.font.family,
+                            x.tick.font.size = 10,
+                            x.tick.label.wrap = TRUE,
+                            x.tick.label.wrap.nchar = 21,
+                            hovertext.font.family = global.font.family,
+                            hovertext.font.size = 11,
+                            marker.size = 6,
+                            swap.x.and.y = FALSE,
+                            legend.bubbles.show = TRUE,
+                            label.auto.placement = TRUE)
+{
+    orig.x <- x
+    checkDataIsEnough(x, y)
+
+    # Try to store name of variables
+    scatter.mult.yvals <- isTRUE(attr(x, "scatter.mult.yvals"))
+    if (!is.null(scatter.sizes) && is.null(scatter.sizes.name))
+        scatter.sizes.name <- deparse(substitute(scatter.sizes))
+    if (!is.null(scatter.labels) && is.null(scatter.labels.name))
+        scatter.labels.name <- deparse(substitute(scatter.labels))
+    if (!is.null(scatter.colors) && is.null(scatter.colors.name))
+        scatter.colors.name <- deparse(substitute(scatter.colors))
+
+    num.tables <- 1
+    groups <- NULL
+    if (is.list(x) && !is.null(ncol(x[[1]])))
+    {
+        output <- unlistX(x, trend.lines)
+        x <- output$x
+        groups <- output$groups
+        num.tables <- output$num.tables
+    }
+
+    x <- convertPercentToProportion(x)
+    annot.data <- x
+
+    if (is.matrix(x) || is.data.frame(x))
```

Comment:
Maybe use `is.array` instead of `is.matrix`? In case the user gives a multi-stat Q table

Reply:
It seems that `unpackColumnsFromX` isn't written to deal with arrays (it would crash?), so `is.matrix` seems correct. This was copied over from `LabeledScatterplot`. How do existing scatterplots handle multi-stat Q table?

Reply:
Hmm...I think scatterplots should be able to deal with multi-stat tables. But maybe that processing happens in PrepareData. I guess its fine to just leave it this way then.